### PR TITLE
AR 3183 optional params fix

### PR DESCRIPTION
**Pull request by makylfang**
- Fixed optional params default value to be complete and replace only items from user input
- previously it had default value only if whole object was missing and not filling defaults of missing keys

### CODE REVIEW

**Comment by shaloo** [comment]
@makylfang this requires a refresh/publish of Auth SDK ref guide, right?

**Comment by makylfang** [comment]
@shaloo generated new docs, also started pointing docs branch to main instead of commit number.

File: docs/README.md [hunk 1]
```diff
@@ -4,9 +4,9 @@ Auth SDK Reference Guide - v0.0.9-beta8 / [Exports](modules.md)
 <a href="#start"><img height="30rem" src="https://raw.githubusercontent.com/arcana-network/branding/main/an_logo_light_temp.png"/></a>
 <h2 align="center"> <a href="https://arcana.network/">Arcana Network Auth SDK </a></h2>
 </p>
-<br/>
+<br>
 <p id="banner" align="center">
-<br/>
+<br>
```
Comment (shaloo) [inline docs/README.md:9]: Docusaurus cribs while compiling if I dont use this form of breaks.

File: docs/README.md [hunk 2]
```diff
@@ -4,9 +4,9 @@ Auth SDK Reference Guide - v0.0.9-beta8 / [Exports](modules.md)
 <a href="#start"><img height="30rem" src="https://raw.githubusercontent.com/arcana-network/branding/main/an_logo_light_temp.png"/></a>
 <h2 align="center"> <a href="https://arcana.network/">Arcana Network Auth SDK </a></h2>
 </p>
-<br/>
+<br>
```
Comment (shaloo) [inline docs/README.md:7]:
```suggestion
<br></br>

File: docs/README.md (hunk 1 above)
Comment (shaloo) [inline docs/README.md:9]:
```suggestion
<br></br>

Comment (makylfang) [reply docs/README.md:9]: @shaloo `<br/>` should work right? I added to dev.

**Pull request merged**